radio/src/gui/colorlcd/model_setup.cpp

```diff
@@ -22,9 +22,91 @@
 #include "opentx.h"
 #include "libopenui.h"
 #include "storage/modelslist.h"
+#include "algorithm"
 
 #define SET_DIRTY()     storageDirty(EE_MODEL)
 
+class RFProtocols
+{
+private:
+    const char * protocols;
+    int maxProtocols;
+    std::vector<std::string> labels;
+    uint8_t getProtoStringLength();
+
+public:
+    RFProtocols(const char * rfProtocols, int max);
+    int findProtocolIndex(int rfProtcolIndex);
+    int lookupProtocolIndex(int selectedIndex);
+    std::vector<std::string> &getLabels();
+};
+
+uint8_t RFProtocols::getProtoStringLength()
+{
+    return protocols[0];
+}
+
+RFProtocols::RFProtocols(const char *protocols, int max)
+{
+    this->protocols = protocols;
+    maxProtocols = max;
+
+    uint8_t len = protocols[0];
+    const char * value = &protocols[1];
+
+    for (int i = 0; i < max; i++)
+    {
+        std::string label = std::string(value, min<uint8_t>(len, strlen(value)));
+        labels.emplace_back(label);
+
+        value += len;
+    }
+    std::sort(labels.begin(), labels.end());
+}
+
+std::vector<std::string> &RFProtocols::getLabels()
+{
+    return labels;
+}
+
+int RFProtocols::lookupProtocolIndex(int selectedIndex)
+{
+    uint8_t len = getProtoStringLength();
+    const char *p = &protocols[1];
+
+    const char *protocolLabel = labels.at(selectedIndex).c_str();
```

Comment:
This line, 77, is where it crashes.

Reply:
i have pushed an update which corrects the crash on selecting protocol 83 / config.  fixed in a different way than what was suggested.  The original code was:

`    for (int i = vmin; i <= vmax; i++) {
      this->values.emplace_back(std::string(value, min<uint8_t>(len, strlen(value))));
      value += len;
    }
`
in my port of this code I had i < max instead of <= max

corrected and pushed.